ui/tests/e2e/settings.spec.js

```diff
@@ -50,6 +50,44 @@ test.describe('Settings Page Tests', () => {
     verifyAdapterResBody(body);
   });
 
+  test('Adapters, Metrics, Registry, Reset tabs are displayed', async ({ page }) => {
+    // Navigate to "Adapters" tab
+    await page.getByRole('tab', { name: 'Adapters' }).click();
+
+    // Mesh Adapter URL and Available Adapters input elements are displayed
+    await expect(page.locator('label').filter({ hasText: 'Mesh Adapter URL' })).toBeVisible();
+    await expect(page.locator('label').filter({ hasText: 'Available Adapters' })).toBeVisible();
+
+    // Navigate to "Metrics" tab
+    await page.getByRole('tab', { name: 'Metrics' }).click();
+
+    // Navigate to "Grafana" sub-tab
+    await page.getByRole('tab', { name: 'Grafana' }).click();
+
+    // Grafana Base URL input elements are displayed
+    await expect(page.locator('label').filter({ hasText: 'Grafana Base URL' })).toBeVisible();
+
+    // Navigate to "Prometheus" sub-tab
+    await page.getByRole('tab', { name: 'Prometheus' }).click();
+
+    // Prometheus Base URL subt-tab are displayed
+    await expect(page.locator('label').filter({ hasText: 'Prometheus Base URL' })).toBeVisible();
+
+    // Navigate to "Registry" tab
+    await page.getByRole('tab', { name: 'Registry' }).click();
```

Comment:
Can we use an id or other static identifier to locate elements other than text that may well change and break the test?

Reply:
Based on the [priority given by Playwright](https://playwright.dev/docs/locators#quick-guide), we can use `getByRole` for querying the DOM. 
And we can use [`getByTestId`](https://playwright.dev/docs/locators#locate-by-test-id) in scenarios where we cannot locate elements by [role](https://playwright.dev/docs/locators#locate-by-role) or [text](https://playwright.dev/docs/locators#locate-by-text).
Additionally, Playwright mentions that we can follow this approach in their [Best Practices section](https://playwright.dev/docs/best-practices#best-practices)

Reply:
Thanks! Regex isn't the solution that we're looking for. It suffers from the same fragility. We're looking for a unique identifier to use as the locator, @senthil-athiban

Reply:
I'll go with unique with the approach of using id as selector.